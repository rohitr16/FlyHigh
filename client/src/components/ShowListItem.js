import React from "react";
import currencyFormatter from "currency-formatter";

function ShowListItem(props = {}) {
  const { data = {}, airportCodeToCityMap = {} } = props;
  const {
    origin: originCode,
    destination: destinationCode,
    departureDate,
    returnDate,
    price = {},
    seatAvailability,
  } = data;
  const { amount, currency } = price;

  const origin = airportCodeToCityMap[originCode] || originCode;
  const destination = airportCodeToCityMap[destinationCode] || destinationCode;

  return (
    <div
      tabIndex="0" // To make it accessible as it will open a details page .
      aria-label="offer card"
      className="card__side card__side--front"
    >
      <h4 className="card__heading">
        <span className="card__heading-span">
          <span className="card__heading-label">{origin}</span>
          <svg
            fill="#000000"
            height="2rem"
            width="2rem"
            version="1.1"
            id="Layer_1"
            xmlns="http://www.w3.org/2000/svg"
            xmlnsXlink="http://www.w3.org/1999/xlink"
            viewBox="0 0 122.88 122.88"
            xmlSpace="preserve"
          >
            <g>
              <path
                fillRule="evenodd"
                clipRule="evenodd"
                d="M16.63,105.75c0.01-4.03,2.3-7.97,6.03-12.38L1.09,79.73c-1.36-0.59-1.33-1.42-0.54-2.4l4.57-3.9
		              c0.83-0.51,1.71-0.73,2.66-0.47l26.62,4.5l22.18-24.02L4.8,18.41c-1.31-0.77-1.42-1.64-0.07-2.65l7.47-5.96l67.5,18.97L99.64,7.45
		              c6.69-5.79,13.19-8.38,18.18-7.15c2.75,0.68,3.72,1.5,4.57,4.08c1.65,5.06-0.91,11.86-6.96,18.86L94.11,43.18l18.97,67.5
		              l-5.96,7.47c-1.01,1.34-1.88,1.23-2.65-0.07L69.43,66.31L45.41,88.48l4.5,26.62c0.26,0.94,0.05,1.82-0.47,2.66l-3.9,4.57
		              c-0.97,0.79-1.81,0.82-2.4-0.54l-13.64-21.57c-4.43,3.74-8.37,6.03-12.42,6.03C16.71,106.24,16.63,106.11,16.63,105.75
		              L16.63,105.75z"
              />
            </g>
          </svg>
          <span className="card__heading-label">{destination}</span>
        </span>
      </h4>
      <div className="card__details">
        {new Date(departureDate).toDateString()} ⇌{" "}
        {new Date(returnDate).toDateString()}
      </div>
      <div className="card__details">
        <span className="card_seat">
          Only {seatAvailability} seats left at →
        </span>
        <span className="card_amount">
          {currencyFormatter.format(amount, { code: currency })}
        </span>
      </div>
    </div>
  );
}

export default ShowListItem;
