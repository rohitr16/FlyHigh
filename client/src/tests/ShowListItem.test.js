import React from "react";
import {cleanup, render, screen} from '@testing-library/react';

import ShowListItem from "../components/ShowListItem";

let container = null;

beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(cleanup);

const data = {
    origin: "DUS",
    destination: "FRA",
    departureDate: "2023-01-14",
    returnDate: "2023-01-17",
    seatAvailability: "7",
    price: {
      amount: 100.96,
      currency: "EUR"
    },
    offerType: "BestPrice",
    uuid: "SA00003-b790715e-b2b8-4d23-ac27-d4RR8c0e84af"
}

const airportCodeToCityMap =  {
    DUS: "Dusseldorf",
    FRA: "Frankfurt"
}

it("renders with  offers", () => {
    const view = render(<ShowListItem  data={data} airportCodeToCityMap={airportCodeToCityMap} />, container);
    expect(screen.getAllByText('Dusseldorf')).toBeTruthy();
    expect(screen.getAllByText('Frankfurt')).toBeTruthy();
    expect(screen.getAllByText('100,96 â‚¬')).toBeTruthy();
    expect(view).toMatchSnapshot();
    
  });
