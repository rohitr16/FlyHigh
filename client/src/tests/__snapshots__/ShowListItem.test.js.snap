// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with  offers 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
    <div>
      <div
        aria-label="offer card"
        class="card__side card__side--front"
        tabindex="0"
      >
        <h4
          class="card__heading"
        >
          <span
            class="card__heading-span"
          >
            <span
              class="card__heading-label"
            >
              Dusseldorf
            </span>
            <svg
              fill="#000000"
              height="2rem"
              id="Layer_1"
              version="1.1"
              viewBox="0 0 122.88 122.88"
              width="2rem"
              xml:space="preserve"
              xmlns="http://www.w3.org/2000/svg"
              xmlns:xlink="http://www.w3.org/1999/xlink"
            >
              <g>
                <path
                  clip-rule="evenodd"
                  d="M16.63,105.75c0.01-4.03,2.3-7.97,6.03-12.38L1.09,79.73c-1.36-0.59-1.33-1.42-0.54-2.4l4.57-3.9 c0.83-0.51,1.71-0.73,2.66-0.47l26.62,4.5l22.18-24.02L4.8,18.41c-1.31-0.77-1.42-1.64-0.07-2.65l7.47-5.96l67.5,18.97L99.64,7.45 c6.69-5.79,13.19-8.38,18.18-7.15c2.75,0.68,3.72,1.5,4.57,4.08c1.65,5.06-0.91,11.86-6.96,18.86L94.11,43.18l18.97,67.5 l-5.96,7.47c-1.01,1.34-1.88,1.23-2.65-0.07L69.43,66.31L45.41,88.48l4.5,26.62c0.26,0.94,0.05,1.82-0.47,2.66l-3.9,4.57 c-0.97,0.79-1.81,0.82-2.4-0.54l-13.64-21.57c-4.43,3.74-8.37,6.03-12.42,6.03C16.71,106.24,16.63,106.11,16.63,105.75 L16.63,105.75z"
                  fill-rule="evenodd"
                />
              </g>
            </svg>
            <span
              class="card__heading-label"
            >
              Frankfurt
            </span>
          </span>
        </h4>
        <div
          class="card__details"
        >
          Sat Jan 14 2023
           ⇌ 
          Tue Jan 17 2023
        </div>
        <div
          class="card__details"
        >
          <span
            class="card_amount"
          >
            100,96 €
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      aria-label="offer card"
      class="card__side card__side--front"
      tabindex="0"
    >
      <h4
        class="card__heading"
      >
        <span
          class="card__heading-span"
        >
          <span
            class="card__heading-label"
          >
            Dusseldorf
          </span>
          <svg
            fill="#000000"
            height="2rem"
            id="Layer_1"
            version="1.1"
            viewBox="0 0 122.88 122.88"
            width="2rem"
            xml:space="preserve"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <g>
              <path
                clip-rule="evenodd"
                d="M16.63,105.75c0.01-4.03,2.3-7.97,6.03-12.38L1.09,79.73c-1.36-0.59-1.33-1.42-0.54-2.4l4.57-3.9 c0.83-0.51,1.71-0.73,2.66-0.47l26.62,4.5l22.18-24.02L4.8,18.41c-1.31-0.77-1.42-1.64-0.07-2.65l7.47-5.96l67.5,18.97L99.64,7.45 c6.69-5.79,13.19-8.38,18.18-7.15c2.75,0.68,3.72,1.5,4.57,4.08c1.65,5.06-0.91,11.86-6.96,18.86L94.11,43.18l18.97,67.5 l-5.96,7.47c-1.01,1.34-1.88,1.23-2.65-0.07L69.43,66.31L45.41,88.48l4.5,26.62c0.26,0.94,0.05,1.82-0.47,2.66l-3.9,4.57 c-0.97,0.79-1.81,0.82-2.4-0.54l-13.64-21.57c-4.43,3.74-8.37,6.03-12.42,6.03C16.71,106.24,16.63,106.11,16.63,105.75 L16.63,105.75z"
                fill-rule="evenodd"
              />
            </g>
          </svg>
          <span
            class="card__heading-label"
          >
            Frankfurt
          </span>
        </span>
      </h4>
      <div
        class="card__details"
      >
        Sat Jan 14 2023
         ⇌ 
        Tue Jan 17 2023
      </div>
      <div
        class="card__details"
      >
        <span
          class="card_amount"
        >
          100,96 €
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
